python/ray/data/_internal/execution/legacy_compat.py

```diff
@@ -137,7 +137,11 @@ def _get_execution_dag(
         record_operators_usage(plan._logical_plan.dag)
 
     # Get DAG of physical operators and input statistics.
-    if DataContext.get_current().optimizer_enabled:
+    if (
+        DataContext.get_current().optimizer_enabled
+        and hasattr(plan, "_logical_plan")
+        and plan._logical_plan is not None
```

Comment:
why is this change needed? add a comment about this?

Reply:
it's because some operators (union and window) doesn't support logical plan yet. Added a comment.